xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/changesinline.vm

```diff
@@ -64,5 +64,9 @@
     #end
   #end
 #end
+#set ($previousOrigdoc = $origdoc.previousVersion)
+#set ($previousNewdoc = $newdoc.previousVersion)
+#set ($nextOrigdoc = $origdoc.documentArchive.getNextVersion($origdoc.RCSVersion))
+#set ($nextNewdoc = $newdoc.documentArchive.getNextVersion($newdoc.RCSVersion))
```

Comment:
It might be a good idea to include the word `version` inside these variable names, otherwise their types can be misunderstood.

For example, `$previousOrigdoc` does not actually contain the previous original document object, but only a version (String or Version object -- it might be a good idea to cast the Version object to String, for consistency).

Note: A good idea at this point would be to observe that we have a `getPreviousVersion` method for a document but we are missing the corresponding `getNextVersion` method. It would be the perfect opportunity to extend the API and add the missing method (to be used when navigating through the document history, as we are doing here), instead of resorting to use the protected API and access the document archive.

Reply:
> A good idea at this point would be to observe that we have a `getPreviousVersion` method for a document but we are missing the corresponding `getNextVersion` method. It would be the perfect opportunity to extend the API and add the missing method

That's actually what I wanted to do but I've been told that a `getNextVersion` method wouldn't be generic enough. What do you think @vmassol?